R/get_ecdc_cases.R

```diff
@@ -28,18 +28,23 @@ get_ecdc_cases <- function(countries = NULL){
   #api may provide file for yesterday's or today's date - it depends on what time it is being accessed
   try_dates <- c(as.Date(Sys.time()) - 1, as.Date(Sys.time()))
 
-  for(i in 1:length(try_dates)){
-    date <- try_dates[i]
+  for (i in 1:length(try_dates)) {
 
+    date <- try_dates[i]
     url <- paste0(base_url, date, '.xls')
+    d <- suppressWarnings(try(tibble::as_tibble(mem_read(url)), silent = TRUE))
 
-    d <- try(tibble::as_tibble(mem_read(url)), silent = TRUE)
+    # if try-error, try again with extension .xlsx
+    if ("try-error" %in% class(d)) {
+      url <- paste0(base_url, date, '.xlsx')
+      d <- suppressWarnings(try(tibble::as_tibble(mem_read(url)), silent = TRUE))
```

Comment:
I saw an error when running this locally - it looks like gdata::read.xls can't handle .xlsx files? 

Of the top of my head not sure of a package that can (and can also download) - I'm not sure `readxl` supports this? Might need to download the file to a temp file and then read in...

Reply:
Oh, it looks like it only handles xlsx if the required perl dependencies are installed. I'll look for a workaround.
https://www.rdocumentation.org/packages/gdata/versions/2.18.0/topics/installXLSXsupport

Reply:
Ah that might be a tricky dependency for people. Can it be automatically installed/flagged - if so could use.

Reply:
Could use `gdata::xlsFormats` to test if system has xlsx capability, and if not direct user to try installing perl dependencies with `gdata::installXLSXsupport`. Or, use `download.file` with `readxl::read_excel`. Any preference?

One challenge with first approach is that there may be multiple reasons why an xlsx file can't be read (e.g. most recent ECDC file is in fact xls, or is xlsx and user is lacking dependencies, or is xlsx and there are other problems).

Reply:
I think the second if possible. I think that will just work without having to get a user to install deps.

Reply:
Ok, I refactored for use `download.file` and `readxl`. I also switched the order of `try_dates` so that it tries the most recent date first. And finally, I added a simple unit test for `get_ecdc_cases`. Note that this will mean the file is downloaded every time a test is run, including on TravisCI. Not sure whether this is desirable.

Happy to make any more changes you suggest, so just let me know

Reply:
Awesome. I guess the only downside (from own end) is that can no longer really use `memoise` but as this shouldn't get hit many times  I don't think it is a problem.